platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java

```diff
@@ -0,0 +1,132 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.email.SmtpClient;
+
+/**
+ * Supports unauthenticated connections to a mail server and username/password authenticated connections
+ * over TLS. If the username is not blank and the password is not empty, then auth/tls will be used.
+ *
+ * The methods {@link #createSession()} and {@link #send(Message)} are thread-safe.
+ *
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public class SmtpClientImpl implements SmtpClient {
+
+    private static final String SMTP_HOST_PROPERTY = "mail.smtp.host";
+
+    private static final String SMTP_PORT_PROPERTY = "mail.smtp.port";
+
+    private static final String SMTP_AUTH_PROPERTY = "mail.smtp.auth";
+
+    private static final String SMTP_START_TLS_ENABLE_PROPERTY = "mail.smtp.starttls.enable";
+
+    private static final String TRUE = "true";
+
+    private static final String FALSE = "false";
+
+    private String hostName;
+
+    private Integer portNumber;
+
+    private String userName;
+
+    private String password;
+
+    /**
+     * Set the username for the email server.
+     *
+     * @param userName may be null or blank
+     */
+    public synchronized void setUserName(String userName) {
+        this.userName = userName;
+    }
+
+    /**
+     * Set the password for the email server.
+     *
+     * @param password may be null or blank
+     */
+    public synchronized void setPassword(String password) {
+        this.password = password;
+    }
+
+    /**
+     * @param hostName must be non-empty
+     */
+    public synchronized void setHostName(String hostName) {
+        notEmpty(hostName, "hostName must be non-empty");
+        this.hostName = hostName;
+    }
+
+    /**
+     * @param portNumber must be non-null and >0
+     */
+    public synchronized void setPortNumber(Integer portNumber) {
+        notNull(portNumber, "portNumber must be non-null");
+        isTrue(portNumber > 0, "portNumber must be >0");
+        this.portNumber = portNumber;
+    }
+
+    @Override
+    public synchronized Session createSession() {
+
+        Properties properties = System.getProperties();
+
+        properties.setProperty(SMTP_HOST_PROPERTY, hostName);
+        properties.setProperty(SMTP_PORT_PROPERTY, portNumber.toString());
+
+        if (StringUtils.isNotBlank(userName) && StringUtils.isNotEmpty(password)) {
+            properties.put(SMTP_AUTH_PROPERTY, TRUE);
+            properties.put(SMTP_START_TLS_ENABLE_PROPERTY, TRUE);
+
+            return Session.getInstance(properties, createAuthenticator());
+        } else {
+            properties.setProperty(SMTP_AUTH_PROPERTY, FALSE);
+            return Session.getInstance(properties);
+        }
+    }
+
+    @Override
+    public synchronized void send(Message message) throws MessagingException {
+        Transport.send(message);
```

Comment:
I didn't think so originally, since the Transport.send javadoc doesn't state that it must be non-null. But I can add a check for it.